clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -1199,6 +1199,168 @@
+
+    public static final Schema ADD_OFFSETS_TO_TXN_REQUEST_V0 = new Schema(
+            new Field("transactional_id",
+                    STRING,
+                    "The transactional id corresponding to the transaction."),
+            new Field("pid",
+                    INT64,
+                    "Current PID in use by the transactional id"),
+            new Field("epoch",
+                    INT16,
+                    "Current epoch associated with the PID"),
+            new Field("consumer_group_id",
+                    STRING,
+                    "Consumer group id whose offsets should be included in the transaction")
+    );
+    public static final Schema ADD_OFFSETS_TO_TXN_RESPONSE_V0 = new Schema(
+            new Field("error_code",
+                    INT16,
+                    "An integer error code.")
+    );
+
+    public static final Schema[] ADD_OFFSETS_TO_TXN_REQUEST = new Schema[] {ADD_OFFSETS_TO_TXN_REQUEST_V0};
+    public static final Schema[] ADD_OFFSETS_TO_TXN_RESPONSE = new Schema[] {ADD_OFFSETS_TO_TXN_RESPONSE_V0};
+
+    public static final Schema END_TXN_REQUEST_V0 = new Schema(
+            new Field("transactional_id",
+                    STRING,
+                    "The transactional id corresponding to the transaction."),
+            new Field("pid",
+                    INT64,
+                    "Current PID in use by the transactional id"),
+            new Field("epoch",
+                    INT16,
+                    "Current epoch associated with the PID"),
+            new Field("transaction_result",
+                    INT8,
+                    "The result of the transaction (0 = COMMIT, 1 = ABORT)")
+    );
+    public static final Schema END_TXN_RESPONSE_V0 = new Schema(
+            new Field("error_code",
+                    INT16,
+                    "An integer error code.")
+    );
+
+    public static final Schema[] END_TXN_REQUEST = new Schema[] {END_TXN_REQUEST_V0};
+    public static final Schema[] END_TXN_RESPONSE = new Schema[] {END_TXN_RESPONSE_V0};
+
+    public static final Schema TXN_MARKER_ENTRY_V0 = new Schema(
+            new Field("pid",
+                    INT64,
+                    "Current PID in use by the transactional id"),
+            new Field("epoch",
+                    INT16,
+                    "Current epoch associated with the PID"),
+            new Field("transaction_result",
+                    INT8,
+                    "The result of the transaction to write to the partitions (0 = COMMIT, 1 = ABORT)"),
+            new Field("topic_partitions",
+                    new ArrayOf(new Schema(
+                            new Field("topic", STRING),
+                            new Field("partitions", new ArrayOf(INT32)))),
+                    "The partitions to write markers for")
+    );
+
+    public static final Schema WRITE_TXN_MARKER_REQUEST_V0 = new Schema(
```

Comment:
This definition differs from the KIP. Is the KIP outdated? If so, can we fix it please? Also, seems like the name should be `WRITE_TXN_MARKERS_REQUEST_V0` since we write multiple markers now.